default/python/AI/AIstate.py

```diff
@@ -74,7 +74,12 @@ def convert_to_version(state, version):
 
     if version == 0:
```

Comment:
> May be make some code that find all function with special name like 'convert_1_2', then detect current code state and current save state and call all functions one by one.

Any particular advantage you see in that approach?
Sure, the if-chain used here may become rather long but it is sequential, self-documenting and logically coherent. It is unambigious and trivial to maneuver to the correct part of the code.
Splitting the body into multiple bodies each called only once seems not to add any kind of real code abstraction.

Particularly long or complex conversion-steps may still be extracted into functions if it seems necessary.

> Maybe move this functions to separate module, so they will not be present in main module. And it will be easy to disable code check for them.

Probably a good idea.